tests/unit/payment-methods/test-class-upe-payment-gateway.php

```diff
@@ -183,84 +192,87 @@ public function set_up() {
 			->disableOriginalConstructor()
 			->getMock();
 
-		$this->mock_payment_methods = [];
-		$payment_method_classes     = [
-			CC_Payment_Method::class,
-			Giropay_Payment_Method::class,
-			Sofort_Payment_Method::class,
-			Bancontact_Payment_Method::class,
-			EPS_Payment_Method::class,
-			P24_Payment_Method::class,
-			Ideal_Payment_Method::class,
-			Sepa_Payment_Method::class,
-			Becs_Payment_Method::class,
-			Link_Payment_Method::class,
+		$this->mock_rate_limiter = $this->createMock( Session_Rate_Limiter::class );
+		$this->order_service     = new WC_Payments_Order_Service( $this->mock_api_client );
+
+		// Arrange: Define a $_POST array which includes the payment method,
+		// so that get_payment_method_from_request() does not throw error.
+		$_POST = [
+			'wcpay-payment-method' => 'pm_mock',
 		];
 
-		$this->mock_rate_limiter = $this->createMock( Session_Rate_Limiter::class );
-		foreach ( $payment_method_classes as $payment_method_class ) {
+		foreach ( $this->payment_method_classes as $payment_method_id => $payment_method_class ) {
 			$mock_payment_method = $this->getMockBuilder( $payment_method_class )
 				->setConstructorArgs( [ $this->mock_token_service ] )
 				->setMethods( [ 'is_subscription_item_in_cart' ] )
 				->getMock();
 			$this->mock_payment_methods[ $mock_payment_method->get_id() ] = $mock_payment_method;
-		}
-
-		$this->order_service = new WC_Payments_Order_Service( $this->mock_api_client );
 
-		// Arrange: Mock UPE_Payment_Gateway so that some of its methods can be
-		// mocked, and their return values can be used for testing.
-		$this->mock_upe_gateway = $this->getMockBuilder( UPE_Payment_Gateway::class )
```

Comment:
For this file, I changed how we test UPE gateways. Instead of using 1 UPE gateway, I create individual UPE gateways for each payment method.

In some test cases, I updated them to loop through all payment methods. In some other test cases, I kept its own testing behaviour by using the Payment_Method::CARD gateway.